pkg/certificate/providers/config.go

```diff
@@ -40,9 +39,9 @@ func NewCertificateManager(kubeClient kubernetes.Interface, kubeConfig *rest.Con
 	// provider generator.
 	mrcClient := &MRCCompatClient{
 		MRCProviderGenerator: MRCProviderGenerator{
-			kubeClient: kubeClient,
-			kubeConfig: kubeConfig,
-			KeyBitSize: cfg.GetCertKeyBitSize(),
+			kubeClient:        kubeClient,
+			certmanagerClient: certmanagerClient,
```

Comment:
why does this need a certmanager client if (in most cases) it will never use it? I'm curious on your motivation to change it

Reply:
Since we [eagerly issue](https://github.com/openservicemesh/osm/pull/4781/files#diff-a79868c58696b375941a0e5a2b1787adab963bde4947a06d3f2402e131ba3c3bR31) a certificate to retrieve the CA during CertificateManager initialization, the tests started to fail since there's no actual cert-manager client, issuer, etc.

Reply:
sounds like that's a symptom of something else. Would you mind reverting, or cutting off a branch/commit that we could look at to see if we can fix the underlying issue without that change?

Reply:
Did some work to reorganize everything and managed to stub out the initial certificate issuance completely in tests. Let me know what you think!